internal/comparetest/compare.go

```diff
@@ -23,19 +23,28 @@ import (
 	"go.uber.org/multierr"
 )
 
-// CompareOption is applied by the CompareMetricSlices function
-// to mutates an expected and/or actual result before comparing.
+// MetricsCompareOption can be used to mutate expected and/or actual metrics before comparing.
+type MetricsCompareOption interface {
+	applyOnMetrics(expected, actual pmetric.Metrics)
+}
+
+// LogsCompareOption can be used to mutate expected and/or actual logs before comparing.
+type LogsCompareOption interface {
+	applyOnLogs(expected, actual plog.Logs)
+}
+
 type CompareOption interface {
-	apply(expected, actual pmetric.Metrics)
+	MetricsCompareOption
+	LogsCompareOption
```

Comment:
Do we need a combined interface?

Reply:
It's returned by `IgnoreResourceAttributeValue` to avoid some code duplication. Otherwise, not needed yet

Reply:
Let me know if you have another suggestion. I can apply that in separate PR